### DISCUSSION TITLE
Auto Generate ID: Option to include custom record ID as a prefix

### DISCUSSION THREAD

**Discussion by WildWombat2000** [burkybang/SuiteAdvanced-Feedback#47] [Ideas]
Would be awesome if the auto-generate ID feature for custom fields had the option to also add (as a prefix) the custom record id (that you're creating the new field for).

e.g. '_customrecord_customfield' instead of just '_customfield'

**Comment by burkybang** [comment]
Interesting idea. However, usually my IDs fill up most if not all of the allocated characters. I feel like most of the time my new custom field would contain most of the ID from the custom record and not much auto generated from the field label.

**Reply by WildWombat2000** [reply]
That happens a lot for me too, however it would be great to have the option there, as when I'm developing a record to be bundled and distributed to other accounts (or even just setting up new records full stop), the record prefix is kinda good to have from an aesthetic and professionalism point of view.

All good if it's not possible, this has already saved me hours in re-typing IDs!

BTW, this is extension is the best thing I have seen for NetSuite in years! Thank you very much for your efforts!

**Reply by burkybang** [reply]
I'm always a fan of options! I'll keep this idea open and let other users upvote it if they'd like it too. That'll let me know how to prioritize it against other tasks.

**Reply by eduardRsult** [reply]
Since the namespace of all custom record custom fields are shared it is actualy very dangerous to not have some prefix there that is specific to the custom record. However because of the character limitation that prefix needs to be as short as possible. Usually for us it is the starting letters of the record type name words.   
Like 

Custom record: Foo Bar. customrecord_eg_foo_bar
an then the field is. Special Date custrecord_eg_fb_special_date
this seems however very specifc.


Another way would be to track the other fields in the record type and if there are 2 fields suggest  when creating the third the consistent part of the first 2. 
Or Since this is mostly only important when you have to create a lot of fields after each other just suggest the last prefix in a memory that is not stored between session.
Not sure if any of this this allignes with how the plugin works.
